paimon-core/src/main/java/org/apache/paimon/io/RowDataFileWriter.java

```diff
@@ -84,6 +85,7 @@ public RowDataFileWriter(
                 DataFileIndexWriter.create(
                         fileIO, dataFileToFileIndexPath(path), writeSchema, fileIndexOptions);
         this.fileSource = fileSource;
+        this.isExternalPath = false;
```

Comment:
We don't need to introduce this bool now, remove it.

Reply:
ok